%\VignetteIndexEntry{DrugDiseaseNet}
%\VignetteDepends{igraph, ROntoTools, graph}
%\VignetteKeywords{Drug Repurposing, Systems biology, Pathway Analysis}
%\VignettePackage{DrugDiseaseNet}
\documentclass[11pt]{article}

\usepackage{hyperref}
\usepackage[margin=1in]{geometry}

\newcommand{\Rfunction}[1]{{\texttt{#1}}}
\newcommand{\Rmethod}[1]{{\texttt{#1}}}
\newcommand{\Robject}[1]{{\texttt{#1}}}
\newcommand{\Rpackage}[1]{{\textit{#1}}}
\newcommand{\Rclass}[1]{{\textit{#1}}}

\begin{document}

\title{DrugDiseaseNet: A package for system-level drug repurposing}
\author{Azam Peyvandipour and Sorin Draghici\\
Department of Computer Science, Wayne State University, Detroit MI 48202}
\maketitle

\begin{abstract}
This package generates a drug-disease network (DDN)
composed of the genes that are most likely perturbed
by a drug. By performing a system-level analysis on this
network using disease gene expression signatures and
drug-exposure gene expression signatures, the amount
of perturbation caused by a drug on the genes can be
estimated. This can help to identify the associations to
a disease of interest. Although such network is studied
in the context of drug repurposing, it also can be used
to identify novel targets for FDA-approved drugs and
understanding their mechanism of action.

\end{abstract}

\section{DrugDiseaseNet}
This document provides an example code that describes
the usage of the package DrugDiseaseNet. This package
uses two different sources of data: one is databases of
pathways and the other is drug targets and disease
related genes. It obtains signaling pathways
from Kyoto Encyclopedia of Genes Genomics (KEGG) \cite{kegg}.
A signaling pathway in KEGG is modeled by a graph in
which nodes represent genes or proteins, and directed
edges between them represent signals between genes
or proteins. The edges are weighted based on the various
types of signals, such as activation, inhibition, etc.
It constructs a global network (GN) by performing
the union of all nodes and edges of KEGG human
signaling pathways.

Drug targets and disease-related genes
(genes associated with the disease of interest)
are retrieved from the Comparative Toxicogenomics
Database (CTD) \cite{mattingly2006comparative}
and Drugbank \cite{wishart2006drugbank}.
CTD is a database that provides curated data describing
cross-species chemical-gene/protein interactions and
gene-disease associations. Drugs with no known targets
are removed from the study. Such drugs are mostly not
FDA-approved.

Next, given the two sets of disease-related genes as
$Disease_t=\{x_1, x_2, ..., x_n\}$, and drug targets as
$Drug_t=\{y_1, y_2, ..., y_n\}$, we extract a subgraph of
$GN$ that consists of all the shortest paths connecting
genes belonging to these sets. It means that a gene from
either $Disease_t$ or $Drug_t$ can be a source or destination
of the shortest path extracted from GN. This subgraph called
Drug-disease network (DDN) represents all the interactions
between drug targets and genes related to the given disease,
through all the interactions described in KEGG signaling pathways.

The impact analysis method \cite{draghici2007systems}
can be applied on DDN using the drug and disease gene
expressions signatures to generate gene perturbation signatures.
The gene perturbation signature is represented by the amount of
perturbation estimated upon genes belonging to the
drug-disease network (DDN) for all drug-disease pairs.

\subsection{KEGG global graph construction }
We design the function \Rmethod{keggGlobalGraph}
to generate the KEGG global graph.
This function uses a copy of KEGG human signaling
pathways. We obtained the KEGG pathways and their
names using the \Rpackage{ROntoTools} package
\cite{voichitaROntoTools}.
The  following code will construct a network using the
available cached data for the human KEGG signaling pathways.
<< message = FALSE,echo = TRUE>>=
library(DrugDiseaseNet)
gg<-keggGlobalGraph()
@

The parameter \textit{updateKEGG} will allow
the user to download the latest KEGG signaling
pathways as follow:

<< message = FALSE,eval=FALSE, echo=TRUE>>=
library(DrugDiseaseNet)
gg<-keggGlobalGraph(updateKEGG=TRUE)
gg
@

At this point, gg is a directed graph of class
\Rclass{graphNEL} with weighted edges.
The edge weights are 1 for activation/expression signals
and -1 for inhibition/repression signals.
The nodes and edges labels can be accessed as follow:

<< message = FALSE,echo = TRUE>>=
library(graph)
allEdges<-edges(gg)
allEdges$"hsa:2065"
head(nodes(gg))
@

The code above shows the edges starting
from node "hsa:2065". The weight of each
edge is obtained using the following code:

<< message = FALSE,echo = TRUE>>=
library(graph)
edgeData(gg,from="hsa:2065",to = "hsa:2549" )
@

Another way to obtain the weight of an edge
is shown as follow:

<< message = FALSE,echo = TRUE>>=
library(graph)
adjmatrix<-as(gg,"matrix")
adjmatrix [1:4,1:4]
adjmatrix ["hsa:2065","hsa:2549"]
@

\subsection{Drug-disease network construction }
In this analysis, the inputs are drug targets,
disease-related genes, and the KEGG global graph.
The output is the directed graph of class
\Rclass{graphNEL} with weighted edges.
This graph is a subgraph of KEGG global
graph connecting drug-targets and disease-related
genes through KEGG signaling pathways.

<< message = FALSE,echo = TRUE>>=
gg<-keggGlobalGraph()
#drug target genes can be obtained from The CTD
drug_targets<-c("hsa:9368","hsa:2322", "hsa:3932", "hsa:4067", "hsa:6714")
#disease-related genes can be obtained from CTD
disease_genes<-c("hsa:1832" ,"hsa:5073" ,"hsa:5328")
drugDiseaseNetwork<-shortestPathsGraph(drug_targets,disease_genes,gg)
drugDiseaseNetwork
@

\section{Funding}
This package was supported in part
by the following grants: NIH R01 DK089167,
NSF 213 DBI-0965741, and R42 GM087013,
and by the Robert J. Sokol Endowment in
Systems Biology.
Any opinions, findings, and conclusions or
recommendations expressed in this material
are those of the authors and do not necessarily
reflect the views of any of the funding agencies.

\section{Citing}
More detail about the proposed approach
is discussed in the manuscript which is in
the publication process.

\bibliographystyle{abbrv}
\bibliography{DrugDiseaseNet.bib}
\end{document}
